app/src/main/java/space/taran/arknavigator/mvp/presenter/ResourcesPresenter.kt

```diff
@@ -359,12 +359,12 @@ class ResourcesPresenter(
     }
 
     private fun initIndexingListeners() {
-        metadataStorage.inProgress.onEach {
+        metadataProcessor.busy.onEach {
             Timber.d("metadata extraction progress = $it")
             viewState.setPreviewGenerationProgress(it)
         }.launchIn(presenterScope)
 
-        previewStorage.inProgress.onEach {
+        previewProcessor.busy.onEach {
             Timber.d("preview generation progress = $it")
             viewState.setPreviewGenerationProgress(it)
         }.launchIn(presenterScope)
```

Comment:
Maybe like this, otherwise one will turn off other
```
viewState.setPreviewGenerationProgress(it || previewProcessor.busy.value)
...
viewState.setPreviewGenerationProgress(it || metadataProcessor.busy.value)
```